packages/widget/src/components/Step/StepTimer.tsx

```diff
@@ -3,11 +3,62 @@ import { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useTimer } from 'react-timer-hook';
 
-const getExpiryTimestamp = (step: Step) =>
-  new Date(
-    (step.execution?.process[0]?.startedAt ?? Date.now()) +
-      step.estimate.executionDuration * 1000,
-  );
+const getExpiryTimestamp = (step: Step) => {
+  const firstProcess = step.execution?.process[0];
+
+  const beginningTimestamp = firstProcess?.approvedAt ?? firstProcess?.doneAt;
+
+  if (!beginningTimestamp) {
+    return new Date(Date.now() + step.estimate.executionDuration * 1000);
+  }
+
+  const timeSpentInExecution =
+    step.execution?.process.reduce((timeConsumed, process, index) => {
+      if (index === 0) {
+        // skip the first process as it is already accounted for
+        return timeConsumed;
+      }
+
+      const { startedAt, doneAt, approvedAt, failedAt } = process;
+
+      // ClonedTimeConsumed is used to include the beginningTimestamp in the calculation
+      let clonedTimeConsumed = timeConsumed;
+
+      if (index === 1) {
+        clonedTimeConsumed += startedAt - beginningTimestamp;
+      }
+      if (process.status !== 'ACTION_REQUIRED') {
+        // if approvedAt is set, then doneAt - approvedAt is the time spent in execution
+        // since startedAt -> approvedAt is the time spent in waiting for the user to confirm the transaction
+
+        if (approvedAt) {
+          if (doneAt) {
+            return clonedTimeConsumed + doneAt - approvedAt;
+          }
+        } else {
+          // if approvedAt is not set, then doneAt - startedAt is the time spent in execution
+          // no user intervention was required
+          if (doneAt && !failedAt) {
+            // if the process failed, then the time for the previous process is not updated.
+            // hence, the time spent in execution will not be considered
+            return clonedTimeConsumed + doneAt - startedAt;
+          }
+        }
+
+        return clonedTimeConsumed;
+      }
+
+      return clonedTimeConsumed;
+    }, 0) ?? 0;
+
+  // timeSpentInExecution is in milliseconds
+  const remainingTimeInSeconds =
+    step.estimate.executionDuration * 1000 - timeSpentInExecution;
```

Comment:
`remainingTimeInSeconds` should be `executionDuration` - `time spent in waiting for required user action`